Sprint 10 ur lrequest

pudramusic opened the pull request.

pudramusic committed:
Изменения
- верстка AuthViewController
- стройтроил AuthViewController в Navigation Controller (через меню Editor — Embed In — Navigation Controller

pudramusic committed:
Изменения
- добавление Alomafire
- создание WebViewViewController и его верстка кодом
- создание перехода между Auth и WebView
- создал URLComponents для создания страницы авторизации и загрузки ее

pudramusic committed:
Изменения
- перенос настроек NavigationBar из WebView в AuthView

pudramusic committed:
Изменения
- добавил WKNavigationDelegate  и провели проверку авторизации пользователя, создав метод webView(_:decidePolicyFor:decisionHandler:), в которую добавляем код, полученный в code(from navigationAction: WKNavigationAction)
- добавил WebViewViewControllerDelegate

pudramusic committed:
Изменения
- изменения по NavigationBar

pudramusic committed:
Изменения:
- исправил все ошибки по NavigationBar. Теперь кнопка "Назад" корректно отображается

pudramusic committed:
Изменения
- Добавил UIProgressView и функцию hideProgressIndicator (но ее нужно пофиксить)

pudramusic committed:
Изменения
- перекинул ProgressView из NavigationBar в webView для корректного отображения. Скорее всего функцию скрытия индикатора придется убрать, так как после сворачивания webView он очищается из памяти

pudramusic committed:
Изменения
- Добавлен в WebViewViewController наблюдатель за ProgressIndicator

pudramusic committed:
Добавил
- сервис OAuth2Service
- библиотеку Alamofire

pudramusic committed:
Изменения
- получен токен через запрос Alamofire
- создана структура для хранения декодинга JSON-ответа
- создано хранилище для токена в UserDefault

pudramusic committed:
Добавил изменения в OAuth2ServiceAlamofire (декодировал данные Json)

pudramusic committed:
Пофиксена функция
extension AuthViewController: WebViewViewControllerDelegate {
    func webViewViewController(_ vc: WebViewViewController, didAuthenticateWithCode code: String)

pudramusic committed:
URL Session + исправление ошибки в первом шаге авторизации - получение кода

pudramusic committed:
Изменение
- Добавлен SplashViewController

andrew-kuzn-yndx commented on imageFeedApp/Auth/AuthViewController.swift:
```diff
@@ -0,0 +1,115 @@
+        configureAuthImageView()
+        configureLoginButton()
+        configureBackButton()
+//        OAuth2Service.shared.fetchToken
```
Старайся не оставлять такое, правилом хорошего тона является чистить код от лишнего перед сдачей работы. В основную ветку проекта должен попадать только чистый код без лишних принтов и закомментированных блоков)

pudramusic replied:
убрал

andrew-kuzn-yndx commented on imageFeedApp/Auth/WebViewViewController.swift:
```diff
@@ -0,0 +1,148 @@
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        webView.removeObserver(self, 
```
Молодец, что предусмотрел удаление Observer!

andrew-kuzn-yndx commented on imageFeedApp/Auth/WebViewViewController.swift:
```diff
@@ -0,0 +1,148 @@
+        if
+            let url = navigationAction.request.url,
+            let urlComponents = URLComponents(string: url.absoluteString),
+            urlComponents.path == "/oauth/authorize/native",
```
Этот path тоже можно поместить в Constamts

pudramusic replied:
перенес

andrew-kuzn-yndx commented on imageFeedApp/Service/Networking/OAuth2Service.swift:
```diff
@@ -0,0 +1,49 @@
+    static let shared = OAuth2Service(); private init() {}
+    
+    private func makeOAuthTokenRequest(code: String) -> URLRequest {
+        guard let baseURL = URL(string: "https://unsplash.com") else {
```
И это в Constants)

pudramusic replied:
перенес

pudramusic committed:
Исправил код, согласно авторскому решению

pudramusic commented:
Исправил проверку токена в Splash (не только nil), надеюсь правильно тебя понял. 
Сделал исправления по незначительным комментариям
Сейчас вроде приложение переходит на экран авторизации, я решил переделать как в Авторском решении, так как в моем варианте перехода не было. Видимо я в функции перехода неверно прописал сам переход
Может нужно было что-то вроде этого, оно и выглядит проще: 
let viewController =AuthViewController()
 splashViewController.pushViewController(viewController, animated: true)
     
 а я намудрил там с UIApplication

pudramusic committed:
Исправлено
- Проверка токена на нил
- Добавлена функция удаления токена. (TODO: Доделать переход после нажатия)

pudramusic committed:
Исправления по сворачиванию WebView. Но не работают

pudramusic committed:
Исправления с Splash

pudramusic committed:
Исправление:
- установил связь с делегатом в AuthViewController, теперь при нажатии кнопки войти меня перекидывает в WebViewView и после логина через делегаты прокидывает в Splash и переключает на флоу с лентой приложения

pudramusic committed:
Удалил лишние принты и настройку Splash. Она оказалось не нужна

pudramusic commented:
Исправил ошибку. Ошибка была в отсутствии связи с делегатом.

pudramusic committed:
Повторная сдача на ревью
